indico/web/client/js/react/components/principals/PersonLinkSearch.jsx

```diff
@@ -0,0 +1,64 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2020 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import {useFavoriteUsers} from 'indico/react/hooks';
+
+import {UserSearch} from './Search';
+
+export const PersonLinkSearch = ({
+  existing,
+  onAddItems,
+  disabled,
+  withExternalUsers,
+  withFavorites,
+  withEventPersons,
+  eventId,
+  single,
+  triggerFactory,
+  ...rest
+}) => {
+  const [favoriteUsers] = useFavoriteUsers();
+  return (
+    <UserSearch
+      existing={existing}
+      onAddItems={onAddItems}
+      favorites={withFavorites ? favoriteUsers : null}
+      disabled={disabled}
+      withExternalUsers={withExternalUsers}
+      withEventPersons={withEventPersons}
+      eventId={eventId}
+      triggerFactory={triggerFactory}
+      single={single}
+      {...rest}
+    />
+  );
+};
+
+PersonLinkSearch.propTypes = {
+  onAddItems: PropTypes.func.isRequired,
+  existing: PropTypes.arrayOf(PropTypes.string).isRequired,
+  disabled: PropTypes.bool,
+  withExternalUsers: PropTypes.bool,
+  withFavorites: PropTypes.bool,
+  withEventPersons: PropTypes.bool,
+  eventId: PropTypes.number,
+  single: PropTypes.bool,
+  triggerFactory: PropTypes.func,
+};
+
+PersonLinkSearch.defaultProps = {
+  disabled: false,
+  withExternalUsers: false,
+  withFavorites: true,
+  withEventPersons: false,
+  eventId: 0,
```

Comment:
`0` is an actual event when you have a DB migrated from legacy, better default to `null`